src/Microsoft.Data/DataFrameTable.cs

```diff
@@ -30,16 +30,10 @@ public DataFrameTable()
         public DataFrameTable(IList<BaseColumn> columns)
         {
             columns = columns ?? throw new ArgumentNullException(nameof(columns));
-            _columns = columns;
-            ColumnCount = columns.Count;
-            if (columns.Count > 0)
+            _columns = new List<BaseColumn>();
```

Comment:
Why duplicate the list now? Why not just hold onto the List that was passed in?

Reply:
2 reasons:
1) I didn't want any external references to the list that the DataFrame is holding onto. This is mostly so that no columns can be added or removed without going through the DataFrame APIs. Considering that the list is only holding references to the columns, I figured this would not be too expensive. It's still not perfect though. The columns themselves could be changed independently, but this would not break the internal book-keeping for the most part
2) `InsertColumn` does some internal validation that I wanted to re-use.